recipes/openblas/all/conanfile.py

```diff
@@ -19,134 +23,221 @@ class OpenblasConan(ConanFile):
+            f_compiler = self._fortran_compiler
+            if not f_compiler:
+                self.output.warning("None or unknown fortran compiler was used.")
+                f_compiler = True
+            self.info.options.build_lapack = f_compiler
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
+        if cross_building(self, skip_x64_x86=True):
             raise ConanInvalidConfiguration("Cross-building not implemented")
 
+        if self.options.use_thread and self.options.use_openmp:
+            raise ConanInvalidConfiguration("Both 'use_thread=True' and 'use_openmp=True' are not allowed")
+
+        if self.settings.os == "Windows" and self.options.build_lapack and self.options.use_openmp:
+            # In OpenBLAS cmake/system.cmake: Disable -fopenmp for LAPACK Fortran codes on Windows
+            self.output.warning("OpenMP is disabled on LAPACK targets on Windows")
+
+    def build_requirements(self):
+        if self.options.build_lapack and self.settings.os == "Windows":
+            self.tool_requires("ninja/1.11.1")
+
+    def system_requirements(self):
+        # TODO: should openmp runtime be system_requirements if 'use_openmp'?
+        # TODO: should gfortran runtime be system_requirement if 'build_lapack' with gfortran compiler
+        # GNU: gomp, LLVM/clang: omp, other: unknown
+        # Debian/Ubuntu: libgomp1, libomp5, Fedora/Rocky/Alpine: libgomp, libomp
+        # Debian/Ubuntu: libgfortran{3,4,5}, Fedora/Rocky/Alpine: libgfortran
+        pass
+
     def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version],
-            strip_root=True,
-            destination=self._source_subfolder
-        )
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
+    def generate(self):
 
+        ninja_generator = "Ninja" if self.settings.os == "Windows" and \
+            self.options.build_lapack else None
+        tc = CMakeToolchain(self, generator=ninja_generator)
+
+        tc.variables["NOFORTRAN"] = not self.options.build_lapack
         if self.options.build_lapack:
-            self.output.warn("Building with lapack support requires a Fortran compiler.")
-        cmake.definitions["NOFORTRAN"] = not self.options.build_lapack
-        cmake.definitions["BUILD_WITHOUT_LAPACK"] = not self.options.build_lapack
-        cmake.definitions["DYNAMIC_ARCH"] = self.options.dynamic_arch
-        cmake.definitions["USE_THREAD"] = self.options.use_thread
+            # This checks explicit user-specified fortran compiler
+            if not self._fortran_compiler:
+                if Version(self.version) < "0.3.21":
+                    self.output.warning(
+                        "Building with LAPACK support requires a Fortran compiler.")
+                else:
```

Comment:
I don't think it's important to guard for different versions here. This modifications in this PR are handled by introducing a new recipe revision. If it can be applied to all versions then it should.
```suggestion
```

Reply:
Since 0.3.21 is the oldest version that allows building LAPACK without Fortran, the recipe needs this guard.